File: experiments/complexity_analysis/On2_quadratic/reference_solutions/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

long long matrixSum(const vector<vector<long long>>& matrix) {
    // O(n²) quadratic time: visit each element exactly once
    long long sum = 0;
    int n = matrix.size();
    
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            sum += matrix[i][j];
        }
    }
    return sum;
}

int main() {
    int n;
    cin >> n;
    
    vector<vector<long long>> matrix(n, vector<long long>(n));
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cin >> matrix[i][j];
        }
    }
    
    // Calculate sum using optimal O(n²) approach
    long long result = matrixSum(matrix);
    cout << result << endl;
    
    return 0;
}

File: experiments/complexity_analysis/On2_quadratic/reference_solutions/solution.py
def matrix_sum(matrix):
    """O(n²) quadratic time: visit each element exactly once"""
    total = 0
    n = len(matrix)
    
    for i in range(n):
        for j in range(n):
            total += matrix[i][j]
    
    return total

# Read input
n = int(input())
matrix = []
for i in range(n):
    row = list(map(int, input().split()))
    matrix.append(row)

# Calculate sum using optimal O(n²) approach
result = matrix_sum(matrix)
print(result)
